torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -106,13 +106,40 @@ template<typename... Args> inline variable_list flatten_tensor_args(Args&&... ar
 
 // See NOTE [ Autograd View Variables ] for details.
 inline Tensor as_view(const Tensor & base, Tensor tensor, bool is_differentiable,
-                      CreationMeta creation_meta=CreationMeta::DEFAULT) {
+        std::function<Tensor(const Tensor&)> func=nullptr,
+        CreationMeta creation_meta=CreationMeta::DEFAULT) {
   auto base_var = Variable(base);
+  // TODO: Potentially the following logic can be moved to VariableType_x.cpp through codegen
+  //       so that we only create chain lambda functions for XLA backend.
   if (base_var.is_view()) {
-    base_var = base_var._base();
+    if (func == nullptr) {
+      // func could be nullptr if tensor is a view created through CreationMeta::MULTI_OUTPUT_NODE.
+      base_var = base_var._base();
+    } else if (base_var.device().type() == at::kXLA) {
+      auto diff_view_meta = static_cast<DifferentiableViewMeta*>(base_var.getIntrusivePtr()->autograd_meta());
```

Comment:
nit: The usual way we get the view meta is: `auto diff_view_meta = static_cast<torch::autograd::DifferentiableViewMeta*>(torch::autograd::impl::get_autograd_meta(self));`